src/integer_mod_q/poly_over_zq/get.rs

```diff
@@ -183,3 +201,54 @@ mod test_get_coeff_z {
         assert_eq!(Z::from(i64::MAX), poly.get_coeff(1).unwrap());
     }
 }
+
+#[cfg(test)]
+mod test_get_degree {
+
+    use crate::integer_mod_q::PolyOverZq;
+    use std::str::FromStr;
+
+    /// ensure that degree is working
+    #[test]
+    fn degree() {
+        let poly = PolyOverZq::from_str("4  0 1 2 3 mod 7").unwrap();
+
+        let deg = poly.get_degree();
+
+        assert_eq!(3, deg);
+    }
+
+    /// ensure that degree is working for constant polynomials
+    #[test]
+    fn degree_constant() {
+        let poly1 = PolyOverZq::from_str("1  1 mod 19").unwrap();
+        let poly2 = PolyOverZq::from_str("0 mod 19").unwrap();
+
+        let deg1 = poly1.get_degree();
+        let deg2 = poly2.get_degree();
+
+        assert_eq!(0, deg1);
+        assert_eq!(-1, deg2);
+    }
+
+    /// ensure that degree is working for polynomials with leading 0 coefficients
+    #[test]
+    fn degree_leading_zeros() {
+        let poly = PolyOverZq::from_str("4  1 0 0 0 mod 199").unwrap();
+
+        let deg = poly.get_degree();
+
+        assert_eq!(0, deg);
+    }
+
+    /// ensure that degree is working for polynomials with many coefficients
+    /// flint does not reduce the exponent due to computational cost
+    #[test]
+    fn degree_many_coefficients() {
+        let poly = PolyOverZq::from_str("7  1 2 3 4 8 1 3 mod 2").unwrap();
```

Comment:
can you make this modulus large, e.g. u64::MAX?
and one with a large coefficient